apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx

```diff
@@ -0,0 +1,132 @@
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import TextareaWithMarkdownPreview from '@cdo/apps/lib/levelbuilder/TextareaWithMarkdownPreview';
+import SaveBar from '@cdo/apps/lib/levelbuilder/SaveBar';
+import {navigateToHref} from '@cdo/apps/utils';
+import $ from 'jquery';
+import color from '@cdo/apps/util/color';
+
+export default function ProgrammingExpressionEditor({
+  initialProgrammingExpression
+}) {
+  const [programmingExpression, setProgrammingExpression] = useState({
+    ...initialProgrammingExpression,
+    id: null,
+    key: null
+  });
+  const [isSaving, setIsSaving] = useState(false);
+  const [lastUpdated, setLastUpdated] = useState(
+    initialProgrammingExpression.lastUpdated
+  );
+  const [error, setError] = useState(null);
+
+  const save = () => {
+    if (isSaving) {
+      return;
+    }
+    setIsSaving(true);
+    fetch(`/programming_expressions/${initialProgrammingExpression.id}`, {
+      method: 'PUT',
+      headers: {
+        'content-type': 'application/json',
+        'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content')
+      },
+      body: JSON.stringify(programmingExpression)
+    })
+      .then(response => {
+        if (response.ok) {
+          setIsSaving(false);
```

Comment:
nit: we always call `setIsSaving(false)` so it can go outside of this conditional